src/components/funding-account/funding-account.service.ts

```diff
@@ -0,0 +1,216 @@
+      .first();
+
+    if (checkFundingAccount) {
+      throw new DuplicateException(
+        'fundingAccount.name',
+        'FundingAccount with this name already exists.'
+      );
+    }
+
+    const secureProps = [
+      {
+        key: 'name',
+        value: input.name,
+        addToAdminSg: true,
+        addToWriterSg: false,
+        addToReaderSg: true,
+        isPublic: false,
+        isOrgPublic: false,
+        label: 'FieldZoneName',
+      },
+    ];
+
+    try {
+      const query = this.db
+        .query()
+        .call(matchRequestingUser, session)
+        .match([
+          node('root', 'User', {
+            id: this.config.rootAdmin.id,
+          }),
+        ])
+        .call(createBaseNode, 'FundingAccount', secureProps)
+        .return('node.id as id');
+
+      const result = await query.first();
+      if (!result) {
+        throw new ServerException('Failed to create funding account');
+      }
+
+      this.logger.info(`funding account created`, { id: result.id });
+
+      return await this.readOne(result.id, session);
+    } catch (err) {
+      this.logger.error('Could not create funding account for user', {
+        exception: err,
+        userId: session.userId,
+      });
+      throw new ServerException('Could not create funding account');
+    }
+  }
+
+  async readOne(id: string, session: ISession): Promise<FundingAccount> {
+    this.logger.info('readOne', { id, userId: session.userId });
+
+    if (!id) {
+      throw new NotFoundException('no id given');
+    }
+
+    if (!session.userId) {
+      session.userId = this.config.anonUser.id;
+    }
+
+    const readFundingAccount = this.db
+      .query()
+      .call(matchRequestingUser, session)
+      .match([node('node', 'FundingAccount', { id })])
+      .call(matchPermList, 'requestingUser')
+      .call(matchPropList, 'permList')
+      .return('propList, permList, node')
+      .asResult<StandardReadResult<DbPropsOfDto<FundingAccount>>>();
+
+    const result = await readFundingAccount.first();
+
+    if (!result) {
+      throw new NotFoundException('FundingAccount.id', 'id');
+    }
+
+    const secured = parseSecuredProperties(
+      result.propList,
+      result.permList,
+      this.securedProperties
+    );
+
+    return {
+      ...parseBaseNodeProperties(result.node),
+      ...secured,
+    };
+  }
+
+  async update(
+    input: UpdateFundingAccount,
+    session: ISession
+  ): Promise<FundingAccount> {
+    const fundingAccount = await this.readOne(input.id, session);
+
+    return await this.db.sgUpdateProperties({
+      session,
+      object: fundingAccount,
+      props: ['name'],
+      changes: input,
+      nodevar: 'fundingAccount',
+    });
+  }
+
+  async delete(id: string, session: ISession): Promise<void> {
+    const fundingAccount = await this.readOne(id, session);
+    try {
+      await this.db.deleteNode({
+        session,
+        object: fundingAccount,
+        aclEditProp: 'canDeleteOwnUser',
```

Comment:
we are in the middle of refactoring how we delete nodes. this function and the test can be 'non-implemented' for now.

Reply:
fyi, we won't be using `acl` concepts at all.

Reply:
Okay